api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -220,18 +230,30 @@ message CommonTlsContext {
   // The same number and types of certificates as :ref:`tls_certificates <envoy_v3_api_field_extensions.transport_sockets.tls.v3.CommonTlsContext.tls_certificates>`
   // are valid in the the certificates fetched through this setting.
   //
-  // If :ref:`tls_certificates <envoy_v3_api_field_extensions.transport_sockets.tls.v3.CommonTlsContext.tls_certificates>`
-  // is non-empty, this field is ignored.
+  // Only one of *tls_certificates*, *tls_certificate_sds_secret_configs*,
+  // and *tls_certificate_provider_instance* may be used.
+  // [#next-major-version: These mutually exclusive fields should ideally be in a oneof, but it's
+  // not legal to put a repeated field in a oneof. In the next major version, we should rework
+  // this to avoid this problem.]
   repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6
       [(validate.rules).repeated = {max_items: 2}];
 
+  // Certificate provider instance for fetching TLS certs.
+  //
+  // Only one of *tls_certificates*, *tls_certificate_sds_secret_configs*,
+  // and *tls_certificate_provider_instance* may be used.
+  // [#not-implemented-hide:]
+  CertificateProviderPluginInstance tls_certificate_provider_instance = 14;
```

Comment:
Now it uses the new type (which is fair since that's the intent) but the field is now called `tls_certificate_provider_instance` and the original field was `tls_certificate_certificate_provider_instance` which combines `tls_certificate` (indicating this is for the identity certificate) and  `certificate_provider_instance` (indicating this is a certificate provider instance). To be really consistent with other fields providing the identity certificate the field would be called `tls_certificate_certificate_provider_plugin_instance` . But I am okay with the name.